paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/CdcRecordStoreWriteOperator.java

```diff
@@ -75,7 +93,7 @@ public void processElement(StreamRecord<CdcRecord> element) throws Exception {
         CdcRecord record = element.getValue();
         Optional<GenericRow> optionalConverted = toGenericRow(record, table.schema().fields());
         if (!optionalConverted.isPresent()) {
-            while (true) {
+            for (int retry = 0; retry < maxRetryNumTimes; ++retry) {
```

Comment:
nit: to me it's slightly unusual to increment the retry counter before entering the block. but i guess it does the same thing as `int retry = 0; retry <= maxRetryNumTimes; retry++` and is just preference